debugclient/getopt.c

```diff
@@ -0,0 +1,1258 @@
+/* Getopt for GNU.
+   NOTE: getopt is now part of the C library, so if you don't know what
+   "Keep this file name-space clean" means, talk to drepper@gnu.org
+   before changing it!
```

Comment:
BSD (old or new), MIT, that's all fine. I think using https://opensource.apple.com/source/Libc/Libc-339/stdlib/FreeBSD/getopt.c (but find an alternative source, not apple's!) should be fine.